streams/src/main/java/org/apache/kafka/streams/processor/DefaultPartitionGrouper.java

```diff
@@ -61,7 +61,7 @@ protected int maxNumPartitions(Cluster metadata, Set<String> topics) {
             List<PartitionInfo> infos = metadata.partitionsForTopic(topic);
 
             if (infos == null)
-                throw new KafkaException("topic not found :" + topic);
+                throw new KafkaException("Topic not found during partition assignment: " + topic);
```

Comment:
That's a good point. Currently the policy is that "anything related to the source / sink topics are thrown with KafkaException since it is out of the blackbox of Streams, and others with StreamsException"; thinking about it a bit more maybe it'd be better to just make it as StreamsException as well.